File: python/data-processing/yaml-config-loader.py
# ---------------------------------------------
# Title: YAML Configuration Loader
# Description: Load multiple YAML config files for Python pipelines
# Input: Config directory with YAML files
# Output: Dictionary with configuration objects
# Source: NBCT_Gene_Analysis
# ---------------------------------------------

import yaml
import os


def load_all_config(config_dir="config"):
    """Load all YAML configuration files."""
    config = {}
    
    config_files = {
        'config': 'config.yaml',
        'columns': 'column_mappings.yaml',
        'cancer_types': 'cancer_types.yaml',
        'params': 'analysis_params.yaml'
    }
    
    for key, filename in config_files.items():
        filepath = os.path.join(config_dir, filename)
        with open(filepath, 'r') as f:
            config[key] = yaml.safe_load(f)
    
    return config


def get_nested_param(config, *keys):
    """Get nested parameter value from config dict."""
    result = config
    for key in keys:
        if key not in result:
            raise KeyError(f"Key not found: {' -> '.join(keys)}")
        result = result[key]
    return result

File: R/tidyverse/yaml-config-loader.R
# ---------------------------------------------
# Title: YAML Configuration Loader
# Description: Load and parse YAML config files for R analysis pipelines
# Input: YAML files in config directory
# Output: List containing configuration objects
# Source: NBCT_Gene_Analysis
# ---------------------------------------------

suppressPackageStartupMessages({
  library(yaml)
})

#' Load main project configuration
load_config <- function(config_dir = "config") {
  config_file <- file.path(config_dir, "config.yaml")
  if (!file.exists(config_file)) {
    stop(paste("Configuration file not found:", config_file))
  }
  yaml.load_file(config_file)
}

#' Load all configuration files at once
load_all_config <- function(config_dir = "config") {
  list(
    config = yaml.load_file(file.path(config_dir, "config.yaml")),
    columns = yaml.load_file(file.path(config_dir, "column_mappings.yaml")),
    cancer_types = yaml.load_file(file.path(config_dir, "cancer_types.yaml")),
    analysis_params = yaml.load_file(file.path(config_dir, "analysis_params.yaml"))
  )
}

#' Get nested parameter value
get_param <- function(params, ...) {
  param_path <- list(...)
  result <- params
  for (key in param_path) {
    if (is.null(resultkey)) {
      stop(paste("Parameter not found:", paste(param_path, collapse = " -> ")))
    }
    result <- resultkey
  }
  result
}
